includes/functions.php

```diff
@@ -2323,56 +2323,72 @@ function pmpro_are_any_visible_levels() {
  */
 function pmpro_getLevelAtCheckout( $level_id = null, $discount_code = null ) {
 	global $pmpro_level, $wpdb, $post;
+	static $pmpro_getLevelAtCheckout_cache = array();
```

Comment:
Have to fix this to avoid warnings when the isset check is made.

Either a better check and setting up the array of arrays on demand. Or concat the level_id . ',' . $discount_code and  hash that and use the hash for the cache key. And/or use the WP get/set cache functions.